cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,221 @@
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil {
+		containerSCC.PrivilegedContainer = *(cut.SecurityContext.Privileged)
+	}
+
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUser = true
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil {
+		containerSCC.ReadOnlyRootFilesystem = *cut.SecurityContext.ReadOnlyRootFilesystem
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil {
+		containerSCC.RunAsNonRoot = *cut.SecurityContext.RunAsNonRoot
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContext = true
+	} else {
+		containerSCC.SeLinuxContext = false
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *v1.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HaveDropCapabilities = false
+	if cut.SecurityContext != nil && cut.SecurityContext.Capabilities != nil {
+		var sliceDropCapabilities []string
+		for _, ncc := range cut.SecurityContext.Capabilities.Drop {
+			sliceDropCapabilities = append(sliceDropCapabilities, string(ncc))
+		}
+		sort.Strings(sliceDropCapabilities)
+		sort.Strings(requiredDropCapabilities)
+		containerSCC.HaveDropCapabilities = reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities)
+		contain := true
+
+		for _, ncc := range cut.SecurityContext.Capabilities.Add {
+			if !contains(category3AddCapabilities, string(ncc)) {
+				contain = false
+			}
+		}
+		if contain {
+			containerSCC.Capabilities = "catagory3"
+		} else {
+			contain = true
+			for _, ncc := range cut.SecurityContext.Capabilities.Add {
+				if !contains(category4AddCapabilities, string(ncc)) {
+					contain = false
+				}
+			}
+			if contain {
+				containerSCC.Capabilities = "catagory4"
+			} else {
+				containerSCC.Capabilities = "catagory5"
+			}
+		}
+	} else {
+		containerSCC.Capabilities = ""
+	}
+	return containerSCC
+}
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func AllVolumeAllowed(volumes []v1.Volume) bool {
+	countVolume := 0
+	for j := 0; j < len(volumes); j++ {
+		if volumes[j].ConfigMap != nil {
+			countVolume++
+		}
+		if volumes[j].DownwardAPI != nil {
+			countVolume++
+		}
+		if volumes[j].EmptyDir != nil {
+			countVolume++
+		}
+		if volumes[j].PersistentVolumeClaim != nil {
+			countVolume++
+		}
+		if volumes[j].Projected != nil {
+			countVolume++
+		}
+		if volumes[j].Secret != nil {
+			countVolume++
+		}
+	}
+	return countVolume == len(volumes)
+}
+
+func Checkcategory(containers []v1.Container, containerSCC ContainerSCC) []string {
+	var badCcontainer []string
+	for j := 0; j < len(containers); j++ {
+		cut := &(containers[j])
+		percontainerSCC := GetContainerSCC(cut, containerSCC)
+		// after building the containerSCC need to check to which category it is
+		switch percontainerSCC {
+		case Category1:
```

Comment:
When doing the comparisons, we should not do a simple "exact" comparison. Maybe we can have a special equal function that would match the scc categories. We should match categories from lowest to highest. If there is not exact match but the container under test parameter needs less privileges than the SCC, it should match. For instance a category3 SCC's compliant container does not need to have privilegedEscalation to be true. If it is false it is also ok.It might need to be SCC level 3 because of other properties, such as capabilities.